File: 2614-prime-in-diagonal/2614-prime-in-diagonal.cpp
class Solution {
public:
    bool check(int a){
        if(a<=1)return false;
        for(int i = 2; i <= sqrt(a); i++){
            if(a%i==0)return false;
        }
        return true;
    }
    int diagonalPrime(vector<vector<int>>& nums) {
        int ans = 0,i,n=nums.size();
        for(i = 0; i < n; i++){
            if(check(nums[i][i])){//CHECK NUMS[I][I] IS PRIME OR NOT
                ans = max(ans,nums[i][i]);
            }
            if(check(nums[i][n-1-i])){//CHECK NUMS[I][N-1-I] IS PRIME OR NOT
                ans = max(ans,nums[i][n-1-i]);
            }
        }
        return ans;
    }
};

File: 2614-prime-in-diagonal/2614-prime-in-diagonal.py
class Solution:
    def diagonalPrime(self, nums: List[List[int]]) -> int:
        
        def isPrime(n):
            if n == 1:
                return False
            i = 2
            while i <= int(math.sqrt(n)):
                if n%i == 0:
                    return False
            
                i += 1
            return True
        
        
        prime = 0
        
        for i in range(len(nums)):
            first = nums[i][i]
            sec = nums[i][len(nums)-i-1]
            
            if isPrime(first):
                prime = max(prime, first)
            
            if isPrime(sec):
                prime = max(prime, sec)
            
        return prime
    
#         def is_prime(n, memo={}):
#             if n < 2:
#                 return False
#             if n in memo:
#                 return memo[n]
            
#             for i in range(2, int(n**0.5) + 1):
#                 if n % i == 0:
#                     memo[n] = False
#                     return False
            
#             memo[n] = True
#             return True